File: Lab7/C++/Lab7.1/Lab7.1/MyStruct.cpp
#include "MyStruct.h"

namespace Lab_7
{

	float Node::getValue() const
	{
		return value;
	}

	void Node::setValue(float value)
	{
		value = value;
	}

	Node *Node::getNext() const
	{
		return next;
	}

	void Node::setNext(Node *value)
	{
		next = value;
	}

	int MyStruct::getSize() const
	{
		return size;
	}

	void MyStruct::setSize(int value)
	{
		size = value;
	}

	Node *MyStruct::getHead() const
	{
		return head;
	}

	void MyStruct::setHead(Node *value)
	{
		head = value;
	}

	Node *MyStruct::getElem() const
	{
		return elem;
	}

	void MyStruct::setElem(Node *value)
	{
		elem = value;
	}

	void MyStruct::push(float val)
	{
		Node *newNode = new Node();
		newNode->setValue(val);

		if (getHead() == nullptr)
		{
		  setHead(newNode);
		}
		 else
		 {
		  newNode->setNext(getHead());
		  setHead(newNode);
		 }
	}

	int MyStruct::count()
	{
	Node *temp = getHead();
	int n = 0;
	float sum = 0;
	while (temp != nullptr)
	{
		sum += temp->getValue();
		temp = temp->getNext();
		if (temp->getValue() > (sum / (getSize() + 1)))
		{
					n += 1;
		}
				temp = temp->getNext();
	}
			return n;
	}

	void MyStruct::exchange()
	{
		Node *temp = getHead();
		while (temp != nullptr)
		{
			if (temp->getValue() < 0)
			{
				temp->setValue(0);
			}
			temp = temp->getNext();
		}
	}

	void MyStruct::output()
	{
		Node *temp = getHead();
		while (temp != nullptr)
		{

			std::wcout << temp->getValue() << L" ";
			temp = temp->getNext();
		}
	}
}

File: Lab7/C#/MyStruct.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using static System.Console;

namespace Lab_7
{
    internal class Node
    {
        public float value { get; set; }
        public Node next { get; set; }
    }
    class MyStruct
    {
        private int size { get; set; }
        private Node head { get; set; }
        private Node elem { get; set; }
        public void push(float val)
        {
            Node newNode = new Node() { value = val };

            if (head == null)
            {
              head = newNode; 
            }
             else
              {
              newNode.next = head;
              head = newNode; 
               }
        }

public int count()  
{  
    Node temp = head;  
    int n = 0;
    float sum = 0;  
    while (temp != null)  
    {  
        sum += temp.value;  
        temp = temp.next; 
        if (temp.value > (sum/(size+1)))
                {
                    n += 1;
                }
                temp = temp.next; 
    }  
            return n;
}  


        public void exchange()
        {
            Node temp = head;
            while (temp != null)
            {
                if (temp.value < 0)
                {
                    temp.value = 0;
                }
                temp = temp.next;
            }
        }

        public void output()
        {
            Node temp = head;
            while (temp != null)
            {
                Write(temp.value.ToString() + " ");
                temp = temp.next;
            }
        }
    }
}